activesupport/lib/active_support/inflector/transliterate.rb

```diff
@@ -68,26 +68,26 @@ def transliterate(string, replacement = "?".freeze)
     #
     #   parameterize("Donald E. Knuth") # => "donald-e-knuth"
     #   parameterize("^trés|Jolie-- ")  # => "tres-jolie"
-    def parameterize(string, sep = '-')
+    def parameterize(string, separator: '-', preserve_case: false)
       # replace accented chars with their ascii equivalents
       parameterized_string = transliterate(string)
       # Turn unwanted chars into the separator
-      parameterized_string.gsub!(/[^a-z0-9\-_]+/i, sep)
-      unless sep.nil? || sep.empty?
-        if sep == "-".freeze
+      parameterized_string.gsub!(/[^a-z0-9\-_]+/i, separator)
+      unless separator.nil? || separator.empty?
```

Comment:
seems like `sep = nil` is somehow supported here, so what I said above won't work here. Instead, we can do:

```ruby
def parameterize(sep = :unused, separator: '-', preserve_case: false)
  unless sep == :unused
    ActiveSupport::Deprecation.warn("Passing a separator as the first argument to `parameterize` is deprecated and will be removed in Rails 5.1. Use the `separator` keyword argument instead, e.g. `name.parameterize(separator: '_')`.")
    separator = sep
  end

  ...
```

I guess we might as well do the same thing in both places for consistency.

If you want to be extra extra sure, you can always use a "poison object", like so:

```ruby
UNUSED = Object.new

def parameterize(sep = UNUSED, separator: '-', preserve_case: false)
  unless sep == UNUSED
    ...
```

But it is probably an overkill here :smile: